lib/auth/sessions.go

```diff
@@ -363,6 +363,17 @@ func (a *Server) newWebSession(
 	if err != nil {
 		return nil, nil, trace.Wrap(err)
 	}
+
+	if tdr, err := a.calculateTrustedDeviceMode(ctx, func() ([]types.Role, error) {
+		return checker.Roles(), nil
+	}); err != nil {
+		log.
+			WithError(err).
+			Warnf("Failed to calculate trusted device mode for session")
+	} else {
+		sess.SetTrustedDeviceRequirement(tdr)
```

Comment:
Note that this is now going to be persisted along with the session, so we should probably update the comments on the proto field.